jaas/src/main/java/com/griddynamics/jagger/jaas/service/DynamicDataService.java

```diff
@@ -0,0 +1,150 @@
+    
+    @PostConstruct
+    public void init() {
+        LOGGER.info("Registering default jagger test db config: {}", defaultJaggerTestDbConfig);
+        jaasDao.createOrUpdate(defaultJaggerTestDbConfig);
+        getDataServiceFor(defaultJaggerTestDbConfig.getName());
+    }
+    
+    @PreDestroy
+    public void destroy() {
+        destroyerService.shutdown();
+    }
+    
+    protected void evictDataServiceFor(final String configName) {
+        LOGGER.info("Evicting jagger test db config with name: {}", configName);
+        AbstractApplicationContext context = dataServiceContexts.remove(configName);
+        if (context != null) {
+            destroyerService.execute(context::destroy);
+        }
+    }
+    
+    public DataService getDataServiceFor(final String configName) {
+        
+        Objects.requireNonNull(configName);
+        
+        ApplicationContext dataServiceContext = dataServiceContexts.get(configName);
+        if (Objects.isNull(dataServiceContext)) {
+            JaggerTestDbConfig config = jaasDao.read(configName);
+            if (Objects.isNull(config)) {
+                return null;
+            }
+            synchronized (this) {
+                dataServiceContext = dataServiceContexts.computeIfAbsent(configName, s -> {
+                    if (dataServiceContexts.size() >= dataServiceCacheSize) {
+                        evictDataServiceFor(dataServiceContexts.keySet().iterator().next());
+                    }
+                    return initDataServiceContextFor(config);
+                });
+            }
+        }
+        
+        return dataServiceContext.getBean(DataService.class);
+    }
+    
+    protected AbstractApplicationContext initDataServiceContextFor(JaggerTestDbConfig config) {
+    
+        LOGGER.debug("Initializing spring context for jagger test db config: {}", config);
+
+        PropertiesPropertySource propertySource =
+                new PropertiesPropertySource(this.getClass().getName(), extractPropsFrom(config));
+        
+        AnnotationConfigApplicationContext applicationContext = new AnnotationConfigApplicationContext();
+        applicationContext.getEnvironment().getPropertySources().addFirst(propertySource);
+        applicationContext.register(DataServiceConfig.class);
+        applicationContext.refresh();
+
+        LOGGER.debug("Spring context has been initialized for jagger test db config: {}", config);
+        return applicationContext;
+    }
+    
+    public Properties extractPropsFrom(JaggerTestDbConfig config) {
+        Properties configProps = new Properties();
+        for (Field field : config.getClass().getDeclaredFields()) {
+            JaggerPropertyName propertyName = field.getAnnotation(JaggerPropertyName.class);
+            if (Objects.nonNull(propertyName)) {
+                field.setAccessible(true);
+                try {
+                    configProps.setProperty(propertyName.value(), (String) field.get(config));
```

Comment:
It's better to use org.springframework.util.ReflectionUtils.setField() here to avoid try-catch